core/governance/market.go

```diff
@@ -611,7 +607,56 @@ func validateNewMarketChange(
 	return types.ProposalErrorUnspecified, nil
 }
 
-func validateParentMarket(prop *types.NewMarket, parent *types.Market) (types.ProposalError, error) {
+func validateSuccessorMarket(terms *types.NewMarket, parent *types.Market) (types.ProposalError, error) {
+	suc := terms.Successor()
+	if parent == nil && suc == nil {
+		return types.ProposalErrorUnspecified, nil
+	}
+	// if parent is not nil, then terms.Successor() was not nil and vice-versa. Either both are set or neither is.
+	if perr, err := validateInsurancePoolFraction(suc.InsurancePoolFraction); err != nil {
+		return perr, err
+	}
+	if perr, err := validateParentProduct(terms, parent); err != nil {
+		return perr, err
+	}
+	// ensure price monitoring parameters are either set, or inherited:
+	if terms.Changes.PriceMonitoringParameters == nil && parent.PriceMonitoringSettings != nil && parent.PriceMonitoringSettings.Parameters != nil {
+		terms.Changes.PriceMonitoringParameters = parent.PriceMonitoringSettings.Parameters.DeepClone()
+	}
+	// same, but ensure the risk parameters are not nil
+	if terms.Changes.RiskParameters == nil && parent.TradableInstrument.RiskModel != nil {
+		switch rm := parent.TradableInstrument.RiskModel.(type) {
+		case *types.TradableInstrumentSimpleRiskModel:
+			terms.Changes.RiskParameters = &types.NewMarketConfigurationSimple{
+				Simple: rm.SimpleRiskModel.Params.DeepClone(),
+			}
+		case *types.TradableInstrumentLogNormalRiskModel:
+			terms.Changes.RiskParameters = &types.NewMarketConfigurationLogNormal{
+				LogNormal: rm.LogNormalRiskModel.DeepClone(),
+			}
+		}
+	}
+	return types.ProposalErrorUnspecified, nil
+}
+
+func validateParentProduct(prop *types.NewMarket, parent *types.Market) (types.ProposalError, error) {
+	assets, err := parent.GetAssets()
```

Comment:
can we add verification that the product is a future product?

Reply:
added a check to make sure both successor and parent are future markets